src/lib.rs

```diff
@@ -281,76 +372,44 @@ impl<'a> Clay<'a> {
         Clay_SetMeasureTextFunction(Some(callback), user_data);
     }
 
-    /// Generates a unique ID based on the given `label`.
-    ///
-    /// This ID is global and must be unique across the entire scope.
-    #[inline]
-    pub fn id(&self, label: &'a str) -> id::Id {
-        id::Id::new(label)
-    }
-
-    /// Generates a unique indexed ID based on the given `label` and `index`.
-    ///
-    /// This is useful when multiple elements share the same label but need distinct IDs.
-    #[inline]
-    pub fn id_index(&self, label: &'a str, index: u32) -> id::Id {
-        id::Id::new_index(label, index)
-    }
-
-    /// Generates a locally unique ID based on the given `label`.
-    ///
-    /// The ID is unique within a specific local scope but not globally.
-    #[inline]
-    pub fn id_local(&self, label: &'a str) -> id::Id {
-        id::Id::new_index_local(label, 0)
-    }
-
-    /// Generates a locally unique indexed ID based on the given `label` and `index`.
-    ///
-    /// This is useful for differentiating elements within a local scope while keeping their labels consistent.
-    #[inline]
-    pub fn id_index_local(&self, label: &'a str, index: u32) -> id::Id {
-        id::Id::new_index_local(label, index)
-    }
-
     /// Sets the maximum number of element that clay supports
     /// **Use only if you know what you are doing or your getting errors from clay**
-    pub fn max_element_count(&self, max_element_count: u32) {
+    pub fn max_element_count(&mut self, max_element_count: u32) {
         unsafe {
             Clay_SetMaxElementCount(max_element_count as _);
         }
     }
     /// Sets the capacity of the cache used for text in the measure text function
     /// **Use only if you know what you are doing or your getting errors from clay**
-    pub fn max_measure_text_cache_word_count(&self, count: u32) {
+    pub fn max_measure_text_cache_word_count(&mut self, count: u32) {
         unsafe {
             Clay_SetMaxElementCount(count as _);
         }
     }
 
     /// Enables or disables the debug mode of clay
-    pub fn enable_debug_mode(&self, enable: bool) {
+    pub fn enable_debug_mode(&mut self, enable: bool) {
         unsafe {
             Clay_SetDebugModeEnabled(enable);
         }
     }
 
     /// Sets the dimensions of the global layout, use if, for example the window size you render to
     /// changed
-    pub fn layout_dimensions(&self, dimensions: Dimensions) {
+    pub fn layout_dimensions(&mut self, dimensions: Dimensions) {
         unsafe {
             Clay_SetLayoutDimensions(dimensions.into());
         }
     }
     /// Updates the state of the pointer for clay. Used to update scroll containers and for
     /// interactions functions
-    pub fn pointer_state(&self, position: Vector2, is_down: bool) {
+    pub fn pointer_state(&mut self, position: Vector2, is_down: bool) {
```

Comment:
Why is this changed to mutable?

Reply:
because it mutates the internal state of clay

Reply:
That is hidden from the perspective of Rust tho.

Reply:
it's to make sure that there's not 2 places in the code updating the pointer state at the same time causing data races